apps/charterafrica/src/lib/data/common/processPageDatasets.js

```diff
@@ -1,21 +1,39 @@
 import fetchDatasets from "@/charterafrica/lib/openAfrica";
+import datasetsQuery from "@/charterafrica/utils/datasets/queryString";
 
-async function processPageDatasets(page, api) {
+const getDatasetsQuery = (context) => {
+  const { query = {}, locale } = context;
+  const { page = 1, q, tags = "", countries = "", sort = "" } = query;
+
+  return { countries, locale, page, q, tags, sort };
+};
+
+export default async function processPageDatasets(page, api, context) {
   const { blocks } = page;
+  const { organizationId } = await api.findGlobal("openAfrica");
+
   const datasetsIndex = blocks.findIndex(({ slug }) => slug === "datasets");
-  if (datasetsIndex > -1) {
-    const { organizationId } = await api.findGlobal("datasets");
-    const datasets = await fetchDatasets(organizationId, {
-      rows: 10,
-      start: 0,
-    });
+
+  if (datasetsIndex > -1 && organizationId) {
+    const data = await fetchDatasets(organizationId);
+    const { count, datasets, countries, tags, totalPages } = data;
 
     blocks[datasetsIndex] = {
-      slug: "datasets",
-      datasets,
+      ...blocks[datasetsIndex],
+      count,
+      countries,
+      data: datasets,
+      tags,
+      totalPages,
+    };
+
+    const swrKey = `/api/v1/resources/datasets`;
+    const qs = datasetsQuery(getDatasetsQuery(context));
+    // eslint-disable-next-line no-param-reassign
+    page.fallback = {
+      [`${swrKey}${qs}`]: [],
```

Comment:
This is still empty (`[]`) Should be something like: https://github.com/CodeForAfrica/ui/blob/96257d7c90e8147c92f45766676c165251d61fe0/apps/charterafrica/src/lib/data/common/processPageArticles.js#L248-L251